game/world/managers/objects/script/ScriptHandler.py

```diff
@@ -0,0 +1,927 @@
+            scripts.append(script.datalong4)
+            weights += (script.dataint4,)
+
+        random_script_id = random.choices(scripts, cum_weights=weights, k=1)[0]
+        
+        self.set_generic_script(script.source, script.target, random_script_id)
+
+        pass
+
+    def handle_script_command_remove_item(self, script):
+        src = None
+        if script.source and not script.target:
+            src = script.source
+        elif script.target and not script.source:
+            src = script.target
+        elif script.source and script.target:
+            src = script.source if script.source.unit_flags & UnitFlags.UNIT_FLAG_PLAYER_CONTROLLED else \
+                script.target if script.target.unit_flags & UnitFlags.UNIT_FLAG_PLAYER_CONTROLLED else None
+
+        if src and src.unit_flags & UnitFlags.UNIT_FLAG_PLAYER_CONTROLLED:
+            src.inventory.remove_items(script.datalong, script.datalong2)
+        else:
+            Logger.warning('ScriptHandler: Neither source nor target are a player, aborting SCRIPT_COMMAND_REMOVE_ITEM')
+
+
+    def handle_script_command_remove_object(self, script):
+        Logger.debug('ScriptHandler: handle_script_command_remove_object not implemented yet')
+        pass
+
+    def handle_script_command_set_melee_attack(self, script):
+        if script.source and script.source.is_alive and script.source:
+            if script.source.has_melee():
+                if script.target and script.target.is_alive:
+                    script.source.attack(script.target)                    
+        else:
+            Logger.warning('ScriptHandler: Invalid source, aborting SCRIPT_COMMAND_SET_MELEE_ATTACK')
+
+        pass
+
+    def handle_script_command_set_combat_movement(self, script):
+        Logger.debug('ScriptHandler: handle_script_command_set_combat_movement not implemented yet')
+        pass
+
+    def handle_script_command_set_phase(self, script):
+        Logger.debug('ScriptHandler: handle_script_command_set_phase not implemented yet')
+        pass
+
+    def handle_script_command_set_phase_random(self, script):
+        Logger.debug('ScriptHandler: handle_script_command_set_phase_random not implemented yet')
+        pass
+
+    def handle_script_command_set_phase_range(self, script):
+        Logger.debug('ScriptHandler: handle_script_command_set_phase_range not implemented yet')
+        pass
+
+    def handle_script_command_flee(self, script):
+        if script.source and script.source.is_alive:
+            if not script.source.unit_flags & UnitFlags.UNIT_FLAG_FLEEING:
+                script.source.unit_flags |= UnitFlags.UNIT_FLAG_FLEEING
+                script.source.set_uint32(UnitFields.UNIT_FIELD_FLAGS, script.source.unit_flags)
+
+                # I feel so dirty doing this but it's not working otherwise.
+                flee_text = self.CREATURE_FLEE_TEXT.male_text
+                flee_text = flee_text.replace('%s ', '')                            
```

Comment:
`%s` should be replaced by the name of the source unit.

Reply:
Yes, but the client doesn't do it even though it should, and if the server does it the text looks like:

Defias Henchman: Defias Henchman attempts to run away in fear!

I went though it with Fluglow and he couldn't find another way either.

Reply:
I see, probably in 0.5.3 it worked differently as the name is always included by default.

Reply:
Early vanilla clients have the same issue. I remove the %s upon loading of broadcast_text if server is compiled for older build.

Reply:
Oh, good to know, I guess we can change it too at the time of loading.

Reply:
I just pushed a commit doing exactly that.

Reply:
> I just pushed a commit doing exactly that.

Great! Can you please add the explanatory comment there too? So we don't forget why we are replacing it on loading? :D

Reply:
Done.